### ISSUE TITLE
Update Javadoc for ShadowContentResolver about deprecation notices

### ISSUE THREAD

**Issue by hoisie** [robolectric/robolectric#8874]
In [ShadowContentResolver](https://github.com/robolectric/robolectric/blob/master/shadows/framework/src/main/java/org/robolectric/shadows/ShadowContentResolver.java) there are a lot of `@Deprecated` methods. However, there are no suggestions to what should be used instead. This has caused a lot of confusion. The Javadoc should be updated with better information.

The alternative to ShadowContentResolver is creating fake ContentProviders and registering them using `Robolectric.setupContentProvider` or `ContentProviderController`.

**Comment by shashankiitbhu** [comment]
Hi, I would like to work on this Issue @hoisie can you assign this to me, thanks

**Comment by shashankiitbhu** [comment]
@utzcoz sure

**Comment by shashankiitbhu** [comment]
HI @utzcoz , should all the ```@Deprecated``` method's javadoc be updated like this?(below code) or unique information should be added for each method

<img width="726" alt="Screenshot 2024-02-27 at 11 12 33 AM" src="https://github.com/robolectric/robolectric/assets/126143257/8d94a12e-ca69-40dd-b244-61aa2f8e946b">

**Comment by hoisie** [comment]
@shashankiitbhu yeah that is on the right track. Something like this could be better:


```java
  /**
   * @deprecated This method affects all calls, and does not work with {@link
   *     android.content.ContentResolver#acquireContentProviderClient}. Instead, use
   *     {@link org.robolectric.Robolectric#setupContentProvider(Class, String)} to
   *     install a test-specific ContentProvider that can return any Cursor.
   */
   ```
